### PR TITLE
docs(table): simplify first basic example

### LINKED ISSUE #6036: [Table] simpler example without example database
#### Bug, feature request, or proposal:
Docs request

#### Motivation

https://github.com/angular/material2/issues/5917#issuecomment-316936589

Also, many of the table-related questions posted here and on stack overflow show that folks don't really understand the purpose and use of ExampleDatabase. They are trying to copy/paste/mold it to fit their solution, when it really is just a support class for the example.

#### Proposed Example

I think first-time readers would greatly benefit from seeing a single-emission data source without any frills. Then, once they want to know how to make their table more interesting and dynamic, they can learn about the benefits of Subjects, DAOs, etc in the more complicated examples with a better understanding of how the foundation works.

```ts
interface Pet {
  name: string;
  type: string;
  age: number;
}


@Component({ ... })
export class TableSimpleExample {
  myPets = [
    { name: 'Boots', type: 'Cat', age: 2 },
    ...
  ];

  dataSource: PetDataSource;

  ngOnInit() {
    this.dataSource = new PetDataSource(this.myPets);
  }
}


export class PetDataSource extends DataSource<Pet> {

  constructor(private pets: Pet[]) {
    super();
  }

  connect(): Observable<Pet[]> {
    return Observable.of(this.pets);
  }
}
```

### PR DESCRIPTION
**Pull request by andrewseguin**
Fixes #6036

### CODE REVIEW

**Comment by andrewseguin** [comment]
Sorry, first PR was accidently done off my master, duh

File: src/material-examples/table-basic/table-basic-example.ts [hunk 1]
```diff
@@ -79,13 +51,9 @@ export class ExampleDatabase {
  * should be rendered.
  */
 export class ExampleDataSource extends DataSource<any> {
```
Comment (willshowell) [inline src/material-examples/table-basic/table-basic-example.ts]: Comment should be updated to not reference ExampleDatabase

**Comment by tinayuangao** [comment]
@ andrewseguin Could you please check the travis test failures?

**Comment by andrewseguin** [comment]
Tests now pass, thanks

**Comment by tinayuangao** [comment]
@andrewseguin could you please rebase? Thanks

**Pull request merged**